cob_bms_driver/src/power_state_aggregator.py

```diff
@@ -121,18 +123,38 @@ def calculate_time_remaining(self):
  
     def publish(self):
         if self.voltage != None and self.current != None and self.remaining_capacity != None and self.full_charge_capacity != None and self.temperature != None and (rospy.Time.now() - self.last_update) < rospy.Duration(1):
+            power_consumption = self.calculate_power_consumption()
+            relative_remaining_capacity = self.calculate_relative_remaining_capacity()
+            time_remaining = self.calculate_time_remaining()
+
             ps = PowerState()
             ps.header.stamp = self.last_update
             ps.voltage = self.voltage
             ps.current = self.current
-            ps.power_consumption = self.calculate_power_consumption()
+            ps.power_consumption = power_consumption
             ps.remaining_capacity = self.remaining_capacity
-            ps.relative_remaining_capacity = self.calculate_relative_remaining_capacity()
+            ps.relative_remaining_capacity = relative_remaining_capacity
             ps.charging = self.charging
-            ps.time_remaining = self.calculate_time_remaining()
+            ps.time_remaining = time_remaining
             ps.temperature = self.temperature
+
+            bs = BatteryState()
+            bs.header.stamp = self.last_update
+            bs.voltage = self.voltage
+            bs.current = self.current
+            bs.charge = self.remaining_capacity
+            bs.design_capacity = self.full_charge_capacity
+            bs.percentage = relative_remaining_capacity
```

Comment:
`power_state.relative_remaining_capacity` is in the range 0.0 - 100.0. `battery_state.percentage` has the range 0.0 - 1.0
```suggestion
            bs.percentage = relative_remaining_capacity / 100.0
```